src/par_iter/internal.rs

```diff
@@ -90,28 +166,35 @@ pub fn bridge<PAR_ITER,C>(mut par_iter: PAR_ITER,
         fn callback<P>(mut self, mut producer: P) -> C::Result
             where P: Producer<Item=ITEM>
         {
-            let producer_cost = producer.cost(self.len);
-            let cost = self.consumer.cost(producer_cost);
-            bridge_producer_consumer(self.len, cost, producer, self.consumer)
+            if producer.weighted() || self.consumer.weighted() {
```

Comment:
I was imagining something like this. You call a function in another crate like:

```rust
fn pythagoras_computation() -> impl ParallelIterator<Item=f64> {...}
```

and you don't know that they have put some weights on it.